js/privatebin.js

```diff
@@ -2119,7 +2119,11 @@ jQuery.PrivateBin = (function($, RawDeflate) {
             if (typeof response === 'string' && response.length > 0) {
                 const shortUrlMatcher = /https?:\/\/[^\s]+/g;
                 const shortUrl = (response.match(shortUrlMatcher) || []).filter(function(a) {
-                    return URL.canParse(a);
+                    try {
+                        return !!new URL(a);
```

Comment:
> This is a fast and easy alternative to constructing a URL within a [try...catch](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch) block.

https://developer.mozilla.org/en-US/docs/Web/API/URL/canParse_static

Apparently that canParse has a performance benefit, so maybe still use it if possible?
We could include a polyfill so our own code does not need to be uglified just because old browsers do not support it yet? https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/web.url.can-parse.js

Or maybe at least keep a todo somewhere to change it in the future?

Reply:
Yes, thats literally the polyfill used. :-) I added the check to retain canParse use if present.

Reply:
Well you replaced it, you did not use the polyfill. As stated it apparently has a performance benefit so yeah…

Reply:
Ah okay fixed.